integrationTests/multiShard/endOfEpoch/rating/rating_test.go

```diff
@@ -0,0 +1,656 @@
+	roundDuration := 6
+	numShards := 1
+	datas := []testData{
+		{
+			consensusSize:     63,
+			shardSize:         shardSize,
+			metaConsensusSize: 63,
+			metaShardSize:     shardSize,
+			waitingSize:       waitingSize,
+			offlineNodes:      0,
+			epochs:            epochsStandard,
+			roundDuration:     roundDuration,
+			roundsPerEpoch:    rounds,
+			numShards:         numShards,
+		},
+		{
+			consensusSize:     shardSize,
+			shardSize:         shardSize,
+			metaConsensusSize: shardSize,
+			metaShardSize:     shardSize,
+			waitingSize:       waitingSize,
+			offlineNodes:      0,
+			epochs:            epochsStandard,
+			roundDuration:     roundDuration,
+			roundsPerEpoch:    rounds,
+			numShards:         numShards,
+		},
+		{
+			numShards:         numShards,
+			consensusSize:     63,
+			shardSize:         shardSize,
+			metaConsensusSize: 63,
+			metaShardSize:     shardSize,
+			offlineNodes:      1,
+			epochs:            epochsStandard,
+			roundDuration:     roundDuration,
+			roundsPerEpoch:    rounds,
+			waitingSize:       waitingSize,
+		},
+		{
+			numShards:         numShards,
+			consensusSize:     shardSize,
+			shardSize:         shardSize,
+			metaConsensusSize: shardSize,
+			metaShardSize:     shardSize,
+			offlineNodes:      1,
+			epochs:            epochsStandard,
+			roundDuration:     roundDuration,
+			roundsPerEpoch:    rounds,
+			waitingSize:       waitingSize,
+		},
+	}
+	return datas
+}
+
+func getValidatorInfos(nodesMap map[uint32][]*integrationTests.TestProcessorNode) map[uint32][]*state.ValidatorInfo {
+	metachainNode := nodesMap[core.MetachainShardId][0]
+	metaValidatoStatisticProcessor := metachainNode.ValidatorStatisticsProcessor
+	currentRootHash, _ := metaValidatoStatisticProcessor.RootHash()
+	validatorInfos, _ := metaValidatoStatisticProcessor.GetValidatorInfoForRootHash(currentRootHash)
+	return validatorInfos
+}
+
+func setOfflineNodes(nodesMap map[uint32][]*integrationTests.TestProcessorNode, offlineNodesCurrent int) map[uint32][][]byte {
+	offlineNodesPks := make(map[uint32][][]byte)
+	shardValidators, _ := nodesMap[core.MetachainShardId][0].NodesCoordinator.GetAllEligibleValidatorsPublicKeys(0)
+	shardWaiting, _ := nodesMap[core.MetachainShardId][0].NodesCoordinator.GetAllWaitingValidatorsPublicKeys(0)
+
+	for i := range shardValidators {
+		shardValidators[i] = append(shardValidators[i], shardWaiting[i]...)
+	}
+
+	for shardId, shardPks := range shardValidators {
+		sort.Slice(shardPks, func(i, j int) bool {
+			return bytes.Compare(shardPks[i], shardPks[j]) < 0
+		})
+
+		for i := 0; i < offlineNodesCurrent; i++ {
+			offlineNodesPks[shardId] = append(offlineNodesPks[shardId], shardPks[i])
+		}
+	}
+	return offlineNodesPks
+}
+
+func writeValidatorInfos(folderName string, epoch int, validatorInfos map[uint32][]*state.ValidatorInfo) {
+	f, _ := os.Create(fmt.Sprintf("%s/validatorInfos_%d.csv", folderName, epoch))
+	defer func() {
+		_ = f.Close()
+	}()
+
+	headerCSV :=
```

Comment:
there are specialized golang structs for writing/reading csv files. Take a look at encoding/csv package

Reply:
fixed. Used csv package.